cdap-tms/src/main/java/co/cask/cdap/messaging/TopicMetadata.java

```diff
@@ -29,19 +29,26 @@
  */
 public class TopicMetadata {
 
-  private static final String TTL_KEY = "ttl";
+  public static final String TTL_KEY = "ttl";
 
   private final TopicId topicId;
   private final Map<String, String> properties;
+  private final boolean validated;
```

Comment:
the validated field is not really part of the equals check. Two metadata are the same if they have the same topicId and properties, whether the properties is validated or not doesn't matter.